grid-proxy/internal/gpuindexer/gpuindexer.go

```diff
@@ -0,0 +1,176 @@
+package gpuindexer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/internal/explorer/db"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/pkg/types"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/direct"
+	rmbTypes "github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/direct/types"
+)
+
+const workerCleanupInterval = 10 * time.Second
```

Comment:
Does this mean the worker dies after 10 seconds? or that's the wait time before bucketing the work units?

Reply:
That's the buffer timeout, the interval in which the buffer will be emptied